test/built-ins/Array/prototype/includes/length-zero-returns-false.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.includes
+description: Returns false if length is 0
+info: |
+  22.1.3.11 Array.prototype.includes ( searchElement [ , fromIndex ] )
+
+  ...
+  2. Let len be ? ToLength(? Get(O, "length")).
+  3. If len is 0, return false.
+  ...
+---*/
+
+var fromIndex = {
+  valueOf: function() {
+    throw new Test262Error();
+  }
+};
+
+var sample = [];
+assert.sameValue(sample.includes(0), false, "returns false");
+assert.sameValue(sample.includes(), false, "returns false - no arg");
+assert.sameValue(
+  sample.includes(0, fromIndex), false,
+  "length is checked before ToInteger(fromIndex)"
```

Comment:
This message will never be displayed to the consumer. I'd recommend either initializing the `Test262Error` with a description string, or (better still) getting rid of the `Test262Error` altogether and instead asserting the value of an invocation counter